### ISSUE TITLE
Intero neovim issues / early adopter feedback

### ISSUE THREAD

**Issue by HolgerPeters** [parsonsmatt/intero-neovim#9]
This might be a bug report (or a usability issue) it might also be just me approaching intero-neovim wrongly.
- I have a project managed by stack (i.e. stack build, etc works).
- I installed intero with `stack install intero`, it reports as 
```
        % intero --version
        0.1.15
```

- I have installed intero-neovim via Plug, removing all other plugins so that I can rule out any interference of plugins:
```
call plug#begin()
Plug 'parsonsmatt/intero-neovim'
call plug#end()
```

I open a file with `nvim src/Foo.hs`

  - When I try to write the file `:w`, I see a first error: `E382: Cannot write, 'buftype' option is set`. This happens 
So far my problems, what works?
  - Trying to run `:InteroDiagnostics` yields an error `E117: Unknown function: intero#diagnostics#report`
  - `:InteroOpen` opens a pane that shows a ghci session, my package seems to have compiled okay. `:InteroLoadModule` seems successull, and I can inspect a type signature. So it basically works (although it took a while to find out how to use it).

PS: While writing this issue, I experienced sudden memory consumption of the intero process. Not quite sure what was behind that, but  I thought it was worth mentioning:

<img width="541" alt="screen shot 2016-07-07 at 21 07 17" src="https://cloud.githubusercontent.com/assets/4097049/16666194/db1c5be0-4486-11e6-8cec-0eaf6eece995.png">

**Comment by parsonsmatt** [comment]
Thanks for the feedback!

For 1) you're not supposed to do `stack install intero` but instead do `stack build intero` in your projects. This ensures that it's using the same versions as all of your dependencies, GHC, cabal, etc. This plugin handles that automatically.

`InteroDiagnostics` is a holdover from `ghcmod-vim` that I hadn't remembered to delete. Thanks for noticing!

You may want to post the memory use issue on the intero repository.

**Comment by HolgerPeters** [comment]
Allright, thanks. Anything on the buftype issue? I can overwrite that setting easily using `:set buftype=` and make the buffer a file again. Yet I think this is a bug, maybe related to https://github.com/parsonsmatt/intero-neovim/blob/48f4cd21a3801766dea91b2724b6aee3e7086f55/autoload/intero/process.vim#L74 ?

**Comment by parsonsmatt** [comment]
Agreed, it's definitely a bug -- the terminal buffer should be `nofile`, but definitely not your ordinary file buffer. I'll have to do some more digging to figure out why.

**Comment by parsonsmatt** [comment]
The buftype issue is fixed as of d0379a8e4953fe45 :) Thanks for trying it out!

**Comment by parsonsmatt** [comment]
And, the `InteroDiagnostics` function reference was deleted. I'll go ahead and close this out.
